shell/browser/api/electron_api_in_app_purchase.cc

```diff
@@ -45,6 +65,38 @@ struct Converter<in_app_purchase::Transaction> {
   }
 };
 
+template <>
+struct Converter<in_app_purchase::ProductSubscriptionPeriod> {
+  static v8::Local<v8::Value> ToV8(
+      v8::Isolate* isolate,
+      const in_app_purchase::ProductSubscriptionPeriod&
+          productSubscriptionPeriod) {
+    gin_helper::Dictionary dict = gin::Dictionary::CreateEmpty(isolate);
+    dict.SetHidden("simple", true);
+    dict.Set("numberOfUnits", productSubscriptionPeriod.numberOfUnits);
+    dict.Set("unit", productSubscriptionPeriod.unit);
+    return dict.GetHandle();
+  }
+};
+
+template <>
+struct Converter<in_app_purchase::ProductDiscount> {
+  static v8::Local<v8::Value> ToV8(
+      v8::Isolate* isolate,
+      const in_app_purchase::ProductDiscount& productDiscount) {
+    gin_helper::Dictionary dict = gin::Dictionary::CreateEmpty(isolate);
+    dict.SetHidden("simple", true);
+    dict.Set("identifier", productDiscount.identifier);
+    dict.Set("type", productDiscount.type);
+    dict.Set("price", productDiscount.price);
+    dict.Set("priceLocale", productDiscount.priceLocale);
+    dict.Set("paymentMode", productDiscount.paymentMode);
```

Comment:
```suggestion
    if (productDiscount.paymentMode == SKProductDiscountPaymentModePayAsYouGo) {
      dict.Set("paymentMode", "payAsYouGo");
    } else if (/* ... */) { /* ... */ }
```

You might need to fiddle around with the enum, not sure if you can #include the relevant header here or not.

Reply:
Nice, thanks for the suggestion.
I tried to do it in the C++ land, but the import becomes a bit complicated and I am not sure how to do it properly.
I decided to do the enum switching in Objective-C instead, where the typing has already properly imported 👍